frame/clearing-house/src/integrations/tests.rs

```diff
@@ -118,155 +145,454 @@ impl Default for MarketConfig {
+
+			let config = MarketConfig {
+				asset: asset_id,
+				vamm_config: VammConfig {
+					base_asset_reserves: UNIT * 100,
+					quote_asset_reserves: UNIT * 100_000,
+					peg_multiplier: 1,
+					twap_period: ONE_HOUR,
+				},
+				..Default::default()
+			};
+			assert_ok!(TestPallet::create_market(Origin::signed(ALICE), config));
+			let market_id = Zero::zero();
+			let market = get_market(&market_id);
+			let vamm_state_before = get_vamm(&market.vamm_id);
+
+			assert_ok!(TestPallet::deposit_collateral(Origin::signed(ALICE), USDC, UNIT * 100));
+			assert_ok!(TestPallet::deposit_collateral(Origin::signed(BOB), USDC, UNIT * 100));
+
+			assert_ok!(TestPallet::open_position(
+				Origin::signed(ALICE),
+				market_id,
+				Long,
+				UNIT * 100,
+				0
+			));
+			assert_ok!(TestPallet::open_position(
+				Origin::signed(BOB),
+				market_id,
+				Long,
+				UNIT * 100,
+				0
+			));
+
+			assert_ok!(TestPallet::close_position(Origin::signed(ALICE), market_id));
+			assert_ok!(TestPallet::close_position(Origin::signed(BOB), market_id));
+
+			// Alice closes her position in profit, Bob closes his position in loss
+			// However, since Alice closes her position first, there are no realized losses in the
+			// market yet, so her profits are outstanding
+			let alice_col = get_collateral(&ALICE);
+			let alice_outstanding_profits = get_outstanding_profits(&ALICE);
+			let bob_col = get_collateral(&BOB);
+			assert!(alice_col + alice_outstanding_profits > bob_col);
+			assert_eq!(alice_col + alice_outstanding_profits + bob_col, UNIT * 200);
+
+			assert_ok!(TestPallet::withdraw_collateral(
+				Origin::signed(ALICE),
+				alice_col + alice_outstanding_profits
+			));
+
+			// vAMM is back to its initial state due to path independence
+			let vamm_state_after = get_vamm(&market.vamm_id);
+			assert_eq!(vamm_state_before.base_asset_reserves, vamm_state_after.base_asset_reserves);
+			assert_eq!(
+				vamm_state_before.quote_asset_reserves,
+				vamm_state_after.quote_asset_reserves
+			);
+		})
 	}
-	.build()
-	.execute_with(|| {
-		set_oracle_for(DOT, 1_000);
-		let config = MarketConfig {
-			asset: DOT,
-			vamm_config: VammConfig {
-				base_asset_reserves: UNIT * 100,
-				quote_asset_reserves: UNIT * 10_000,
-				peg_multiplier: 10,
-				twap_period: ONE_HOUR,
-			},
-			margin_ratio_initial: Decimal::from_float(0.1),
-			margin_ratio_maintenance: Decimal::from_float(0.02),
-			margin_ratio_partial: Decimal::from_float(0.04),
-			minimum_trade_size: 0.into(),
-			funding_frequency: ONE_HOUR,
-			funding_period: ONE_HOUR * 24,
-			taker_fee: 0,
-			twap_period: ONE_HOUR,
-		};
-		assert_ok!(TestPallet::create_market(Origin::signed(ALICE), config));
 
-		assert_ok!(TestPallet::deposit_collateral(Origin::signed(BOB), USDC, UNIT * 100));
+	#[test]
+	#[ignore = "FIXME: vAMM TWAP isn't updated if last twap timestamp is equal to the current \
+	block's timestamp"]
+	fn should_update_vamm_twap_in_the_same_block() {
```

Comment:
@Cardosaum this test could be useful for you when evaluating how to update the vAMM's TWAP for intra-block trades